pennylane/transforms/classical_jacobian.py

```diff
@@ -65,39 +72,126 @@ def classical_jacobian(qnode):
 
-    - The second column consisting of all zeros indicates that the generated quantum circuit does
-      not depend on the first element of ``weights``.
+    - The third row consisting of all zeros indicates that the third gate ``RY(2.5)`` does not
+      depend on the ``weights``.
+
+    - The quadratic dependence of the fourth gate argument yields :math:`2\cdot 0.6=1.2`.
+
+    .. note::
+
+        The QNode is constructed during this operation.
+
+    For a QNode with multiple QNode arguments, the arguments with respect to which the
+    Jacobian is computed can be controlled with the ``argnums`` keyword argument.
+    The output for ``argnums=None`` depends on the backend:
+
+    .. list-table:: Output format of ``classical_jacobian``
+       :widths: 25 25 25 25
+       :header-rows: 1
+
+       * - Interface
+         - `argnums=None`
+         - `type(argnums)=int`
+         - argnums=Sequence[int]`
+       * - `'autograd'`
+         - `tuple(arrays)`\*
+         - `array`
+         - `tuple(array)`
+       * - `'jax'`
+         - `array`
+         - `array`
+         - `tuple(array)`
+       * - `'tf'`
+         - `tuple(arrays)`
+         - `array`
+         - `tuple(array)`
+       * - `'torch'`
+         - `tuple(arrays)`
+         - `array`
+         - `tuple(array)`
+
+    **Example with ``argnums``**
+
+    >>> @qml.qnode(dev)
+    ... def circuit(x, y, z):
+    ...     qml.RX(qml.math.sin(x), wires=0)
+    ...     qml.CNOT(wires=[0, 1])
+    ...     qml.RY(y ** 2, wires=1)
+    ...     qml.RZ(1 / z, wires=1)
+    ...     return qml.expval(qml.PauliZ(0)@qml.PauliZ(1))
+    >>> jac_fn = qml.transforms.classical_jacobian(circuit, argnums=[1, 2])
+    >>> x, y, z = np.array([0.1, -2.5, 0.71])
+    >>> jac_fn(x, y, z)
+    (array([-0., -5., -0.]), array([-0.        , -0.        , -1.98373339]))
+
+    Only the Jacobians with respect to the arguments ``x`` and ``y`` were computed, and
+    returned as a tuple of ``arrays``.
+
     """
 
     def classical_preprocessing(*args, **kwargs):
-        """Returns the trainable gate parameters for
-        a given QNode input"""
+        """Returns the trainable gate parameters for a given QNode input."""
         qnode.construct(args, kwargs)
         return qml.math.stack(qnode.qtape.get_parameters())
 
     if qnode.interface == "autograd":
-        return qml.jacobian(classical_preprocessing)
+
+        def _jacobian(*args, **kwargs):
+            if argnums is None:
+                jac = qml.jacobian(classical_preprocessing)(*args, **kwargs)
+            elif np.isscalar(argnums):
+                jac = qml.jacobian(classical_preprocessing, argnum=argnums)(*args, **kwargs)
+            else:
+                jac = tuple(
+                    (
+                        qml.jacobian(classical_preprocessing, argnum=i)(*args, **kwargs)
+                        for i in argnums
+                    )
+                )

```

Comment:
Once we 'fix' `qml.jacobian`, I suppose we can remove the `if` statement here, and simply replace it with
```python
if qnode.interface == "autograd":
    return qml.jacobian(classical_preprocessing, argnum=argnums)
```

Reply:
Yes, I'd think so :) The tests also become a bit simpler :)